install/src/main/java/org/jetbrains/bsp/bazel/install/BloopEnvironmentCreator.kt

```diff
@@ -0,0 +1,71 @@
+package org.jetbrains.bsp.bazel.install
+
+import io.vavr.control.Try
+import org.jetbrains.bsp.bazel.commons.Constants
+import org.jetbrains.bsp.bazel.install.cli.CliOptions
+import org.jetbrains.bsp.bazel.installationcontext.InstallationContext
+import java.nio.file.Path
+import java.nio.file.Paths
+
+private data class WorkspaceSettings(
+    val refreshProjectsCommand: List<String>
+)
+
+private data class ProjectSettings(
+    val targets: List<String>?,
+    val projectName: String?
+)
+
+class BloopEnvironmentCreator(
+    private val cliOptions: CliOptions,
+    private val installationContext: InstallationContext
+) : EnvironmentCreator(cliOptions.workspaceRootDir) {
+
+    private val projectRootDir = cliOptions.workspaceRootDir
+    private val launcherArgumentCreator = LauncherArgumentCreator(installationContext)
+
+    override fun create(): Try<Void> = createDotBazelBsp()
+        .flatMap { bazelBspPath ->
+            createProjectSettings(bazelBspPath).flatMap {
+                BloopBspConnectionDetailsCreator(bazelBspPath).create()
+            }
+        }
+        .flatMap { createDotBsp(it) }
+        .flatMap { createDotBloop() }
+
+    private fun createProjectSettings(bazelBspPath: Path): Try<Void> {
+        val projectSettings = ProjectSettings(
+            cliOptions.projectViewCliOptions?.targets,
+            cliOptions.bloopCliOptions.projectName
+        )
+        val settingsFile = bazelBspPath.resolve("project.settings.json")
+        return writeJsonToFile(settingsFile, projectSettings)
+    }
+
+    private fun createDotBloop(): Try<Void> =
+        createDir(projectRootDir, ".bloop")
+            .flatMap(::createBloopConfig)
+
+    private fun createBloopConfig(path: Path): Try<Void> =
+        refreshProjectArgs().flatMap {
+            val settings = WorkspaceSettings(it)
+            val bloopSettingsJsonPath = path.resolve(Constants.BLOOP_SETTINGS_JSON_FILE_NAME)
+            writeJsonToFile(bloopSettingsJsonPath, settings)
+        }
+
+
+    private fun refreshProjectArgs(): Try<List<String>> {
+        val pwd = Paths.get("").toAbsolutePath()
```

Comment:
so the question why do we want to know this directory? it's a directory where the installer was called - it can be called anywhere

Reply:
the way we've been doing it right now is calling the installer from the root of the repository, so `pwd` == repository root.

Reply:
oh I think there's a difference here.  The "workspace root" is where bazel-bsp writes its files (as well as bloop in this case).  This (for our use case at least) is outside of the "repository root".  We could add another flag for repository root to make this explicit, using the current working directory is a little janky I agree.

Reply:
ohh i see, it seems like a bazel-bsp limitation as well - https://youtrack.jetbrains.com/issue/BAZEL-108

Reply:
Cool, yeah.  With these changes I think it'd be pretty easy now to make the normal bazel-bsp support it too.

Reply:
yup, but i think we can do it in a separate PR